src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -12,9 +14,19 @@
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
 
+  private SecureRandom generator; //
+
+  {
```

Comment:
Move the contents of this initializer to a standard constructor or to field initializers.